src/app/modules/ix-forms/components/ix-combobox/ix-combobox.component.ts

```diff
@@ -124,13 +124,13 @@ export class IxComboboxComponent implements ControlValueAccessor, OnInit {
            * if we haven't found the correct option in the list of options fetched so far. The assumption
            * is that the correct option exists in one of the following pages of list of options
            */
-          this.options.push({ label: this.value as string, value: this.value });
-          this.selectedOption = { ...this.options.find((option: Option) => option.value === this.value) };
-          if (this.selectedOption) {
+          this.selectedOption = { label: this.value as string, value: this.value };
```

Comment:
I believe there was a reason why this was added. Something about having multiple pages of suggestions and having currently selected option on the page 3.
You can check git blame for ticket.
I don't think this functionality needs to be removed for non-custom values.